### PR TITLE
bpo-38185: Fixed case-insensitive string comparison in sqlite3.Row indexing.

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
https://bugs.python.org/issue38185

### CODE REVIEW

**Review by berkerpeksag — APPROVED** [review]
Nice find! The PR looks good to me. I probably wouldn't bother fixing it in 3.7, but it's ultimately up to you :)

File: Modules/_sqlite/row.c [hunk 1]
```diff
@@ -76,16 +76,38 @@ PyObject* pysqlite_row_item(pysqlite_Row* self, Py_ssize_t idx)
...
+equal_ignore_case(PyObject *left, PyObject *right)
+{
+    int eq = PyObject_RichCompareBool(left, right, Py_EQ);
+    if (eq) { /* equal or error */
+        return eq;
+    }
+    if (!PyUnicode_Check(left) || !PyUnicode_Check(right)) {
+        return 0;
+    }
+    if (!PyUnicode_IS_ASCII(left) || !PyUnicode_IS_ASCII(right)) {
```
Comment (ammaraskar) [inline Modules/_sqlite/row.c:89]: Hmm, looks like sqlite supports non-ascii characters as identifiers: https://github.com/mackyle/sqlite/blob/a419afd73a544e30df878db55f7faa17790c01bd/ext/misc/normalize.c#L174-L177

And as shown in your example on bpo, should we maybe just simplify this code and do a `casefold` comparison?

Comment (serhiy-storchaka) [reply Modules/_sqlite/row.c:89]: My initial version used `casefold`. But then I made some experiments and found that Sqlite itself ignores case only for ASCII letters. It was just my guess, and thank you for confirming it with the reference to sources.

**Comment by serhiy-storchaka** [comment]
It could be fixed even in 2.7, but I don't bother. The bug is too tiny.

File: Modules/_sqlite/row.c (hunk 1 above)
Comment (ammaraskar) [reply Modules/_sqlite/row.c:89]: Aah okay, I thought the case-insensitivity for the `Row` object was for the convenience of the Python API, not trying to mirror any particular part of sqlite itself. Which part of sqlite ignores case?

Comment (serhiy-storchaka) [reply Modules/_sqlite/row.c:89]: > Which part of sqlite ignores case?

I guess any part which compare identifiers. For example, the following code passes:
```sql
create table test(a text)
select A from test
```
and the following fails:
```sql
create table test(a text, A text)
```

Comment (ammaraskar) [reply Modules/_sqlite/row.c:89]: Aah gotcha, thanks for taking the time to explain Serhiy.

**Pull request merged**